crates/cli/tests/snapshots/codegen__codegen_csharp.snap

```diff
@@ -672,27 +672,33 @@ namespace SpacetimeDB
 
 			public readonly ref struct IdUniqueIndex
 			{
+				readonly PkMultiIdentityHandle Handle;
```

Comment:
The logic seems unnecessarily complicated.

Table handle stores the Dictionary<> caches.
Table handles also store instances of index structs like this IdUniqueIndex.
Then, index structs store a reference back to the parent table handles.
And then, index structs only access said relevant cache via those table handles.

Instead, let's move the Dictionary<> cache directly into the index struct. This way we'll encapsulate the cache data inside the relevant index struct, and won't need this kind of 3-way access just to retrieve values from the cache from inside the index.

(originally posted at https://github.com/clockworklabs/com.clockworklabs.spacetimedbsdk/pull/168#discussion_r1810863587 )